src/resource_api/schema.py

```diff
@@ -190,8 +190,21 @@ def get_schema(self):
 
 
 class IndexableField(BaseSimpleField):
+    """
+    Base class for most of the primitive fields.
+
+    choices (list|tuple = None)
+        Predefined list of possible values for the field. ValidationError will be raised if field value doesn't
+        match any of the defined values.
+    choice_labels (list|tuple = None)
```

Comment:
I gave your idea another thought.

Human labels just do not make sense with respect to Resource API. The choices that are mentioned in the current "choices" field must be the values that make sense for the client (including a human client). 

The only scenario where you would hypothetically need both: short values and human readable title is in case if you want to "optimize bandwidth" so to say. If performance is to be kept in mind - Resource API is not an option. That has never been its design goal.

In case if you want to implement something like:

```python
number_of_group_members = IntegerFIeld(
    choices=[12, 57, 187],
    choice_labels=["Small group count", "Medium group count", "Big group count"])
```
Please don't.

Do like:

```python
group_type = StringField(choices=["Small", "Medium", "Big"])
```

And internally - i.e. in the implementation (not definition) of the individual resource do the mapping from the string values to the integers.

On the other hand, if in the example above you really need to expose both the size and a corresponding group type as human readable string - it is a clear indication that you should use a read-only Resource with a Link field.

Sth like:

```python

class GroupType(Resource):

    count_name_map = {
        12: "Small",
        57: "Medium",
        187: "Big"
    }

    class Schema:
        count = IntegerField(description="Number of people in a group", pk=True)
        name = StringField(description="Human readable definition of a group")

    def exists(self, user, pk):
        return pk in self.count_name_map

    def get_data(self, user, pk):
        return self.count_name_map.get(pk)

    def get_uris(self, user, params=None):
        return self.count_name_map.keys()

    def get_count(self, user, params=None):
        return len(self.count_name_map)

...
# Somewhere in another resource

Links
    class group_type(Link):
        target = "Teacher"
        one_way = True
        cardinality = Link.cardinalities.ONE
        master = True

```



Thus I do not approve this commit :)